### ISSUE TITLE
'... all-product-headers.yaml' not found

### ISSUE THREAD

**Issue by olafandreas-itf** [facebook/facebook-ios-sdk#1628]
## Checklist

- [x] I've updated to the latest released version of the SDK
- [x] I've searched for existing [GitHub issues](https://github.com/facebook/facebook-ios-sdk/issues)
- [x] I've looked for existing answers on [Stack Overflow](https://facebook.stackoverflow.com), the [Facebook Developer Community Forum](https://developers.facebook.com/community/) and the [Facebook Developers Group](https://www.facebook.com/groups/fbdevelopers)
- [x] I've read the [Code of Conduct](https://github.com/facebook/facebook-ios-sdk/blob/master/CODE_OF_CONDUCT.md)
- [x] This issue is not security related and can safely be disclosed publicly on GitHub

## Environment

Describe your dev environment here, giving as many details as possible. If you have them, make sure to include:

- Xcode Version: `12.4 (12D4e)`
- Swift Version: `5` (if issue is Swift related)
- Installation Platform & Version: `[Cocoapods]` version `1.10.1`

## Goals

What do you want to achieve?
I want to be able to debug my code without being met by errors in the console due to wrongly packaged framework.

## Expected Results

What do you expect to happen?
`po shared` -> should output debug information for the shared object.

## Actual Results

What actually happened? Can you provide a stack trace?
```
po shared
error: virtual filesystem overlay file '/Users/kylinchang/Library/Developer/Xcode/DerivedData/FacebookSDK-axnectjfufffeccdyyliqwafpkfw/Build/Intermediates.noindex/ArchiveIntermediates/FBSDKCoreKit-Dynamic/IntermediateBuildFilesPath/FBSDKCoreKit.build/Release-iphoneos/FBSDKCoreKit-Dynamic.build/all-product-headers.yaml' not found

error: couldn't IRGen expression. Please check the above error messages for possible root causes.
```

## Steps to Reproduce

What are the steps necessary to reproduce this issue?
Installing FacebookSDK/CoreKit (9.0.0) in any project would cause this issue.

## Code Samples & Details

Similar issue https://github.com/Instabug/Instabug-iOS/issues/368

https://bugs.swift.org/browse/SR-12783
>LLDB could clearly do a better job of surfacing the error — I had to dig it out of the types log:
SwiftASTContext("Instabug")::GetASTContext() – failed to initialize ClangImporter: error: virtual filesystem overlay file '/Users/khaledElMorabea/Library/Developer/Xcode/DerivedData/Instabug-ccnahiryvffhsohhjpovgyeazflu/Build/Intermediates.noindex/ArchiveIntermediates/Instabug/IntermediateBuildFilesPath/Instabug.build/Release-iphonesimulator/Instabug.build/all-product-headers.yaml' not found
The problem here is with the binary Instabug framework that you are using. The .dSYM files that are generated from Swift code are not (yet) as portable as the ones generated from Clang code. The binary Swift module encode a hardcoded path to a yaml file that only exists on the original developer's machine. You should let them know that they need to compile their binary framework with -no-serialize-debugging-options if they are planning to distribute them to another machine.

tl;dr
Compile the binary framework with -no-serialize-debugging-options.

**Comment by joesus** [comment]
Oooh interesting. Is this referring to the XCFrameworks specifically? How are you incorporating the SDK?

**Comment by deryck13** [comment]
I have the same issue on version 9.0.0, added to the project with CocoaPods.

**Comment by olafandreas-itf** [comment]
> Oooh interesting. Is this referring to the XCFrameworks specifically? How are you incorporating the SDK?

Not sure if I understand your first question, but I'll have a go:
When not using `-no-serialize-debugging-options` while building frameworks, the package is embedded with absolute paths for search paths etc. from the packaging machine. This would explain the `/Users/kylinchang/Library/...` path I'm seeing, I guess `kylinchang` is the one who packaged the framework for `FacebookSDK/CoreKit`.

So I would guess, Yes.. this refers to XCFrameworks specifically.

For the second question, I'm using CocoaPods.. as stated in the issue template I've filled out.

**Comment by joesus** [comment]
We've added the build flag to the build job for the XCFrameworks so this should be fixed in the next patch release. Thanks for bringing it to our attention!
